priv/perf/apps/load_test/lib/scenario/account_transactions.ex

```diff
@@ -0,0 +1,268 @@
+              amount: @test_output_amount,
+              currency: Encoding.to_hex(@eth),
+              owner: Encoding.to_hex(session.assigned.faucet.addr)
+            }
+          ]
+        }
+      )
+
+    %{
+      "result" => "complete",
+      "transactions" => [
+        %{
+          "inputs" => inputs,
+          "sign_hash" => sign_hash,
+          "typed_data" => typed_data,
+          "txbytes" => txbytes
+        }
+      ]
+    } = Jason.decode!(response.body)["data"]
+
+    {:ok, [inputs, sign_hash, typed_data, txbytes]}
+  end
+
+  defp get_balance(sender) do
+    {:ok, response} =
+      Api.Account.account_get_balance(
+        Connection.client(),
+        %Model.AddressBodySchema{
+          address: Encoding.to_hex(sender.addr)
+        }
+      )
+
+    {:ok, response.body}
+  end
+
+  defp get_utxos(sender) do
+    {:ok, response} =
+      Api.Account.account_get_utxos(
+        Connection.client(),
+        %Model.AddressBodySchema1{
+          address: Encoding.to_hex(sender.addr)
+        }
+      )
+
+    {:ok, response.body}
+  end
+
+  defp get_transactions(_sender) do
+    # There is an issue openapi generated client does not work well with optional request body
+    # https://github.com/OpenAPITools/openapi-generator/issues/5234
+    # So we are not filtering by sender now
+
+    {:ok, response} =
+      Api.Account.account_get_transactions(
+        Connection.client(),
+        %Model.GetAllTransactionsBodySchema{}
+      )
+
+    {:ok, response.body}
+  end
+
+  defp sign_tx(inputs, sign_hash, typed_data, sender) do
+    signature =
+      sign_hash
+      |> Encoding.to_binary()
+      |> Hash.sign_hash(sender.priv)
+      |> Hash.pack_signature()
+      |> Encoding.to_hex()
+
+    signatures = Enum.map(inputs, fn _ -> signature end)
+    Map.put_new(typed_data, "signatures", signatures)
+  end
+
+  defp do_wait_for_balance_update(_session, _sender, 0), do: :wait_for_balance_failed
+
+  defp do_wait_for_balance_update(session, sender, retry) do
+    {:ok, response_body} = get_balance(sender)
+    utxos = Jason.decode!(response_body)["data"]
+
+    if Enum.empty?(utxos) do
+      Process.sleep(@poll_interval)
+
+      session
+      |> log_debug("retry for the balance update for sender: #{Encoding.to_hex(sender.addr)}")
+      |> do_wait_for_balance_update(sender, retry - 1)
+    else
+      {:ok, session}
+    end
+  end
+
+  defp wait_until_tx_sync_to_watcher(session, tx_id) do
+    {:ok, session} = do_wait_until_tx_sync_to_watcher(session, tx_id, @default_retry_attempts)
+    session
+  end
+
+  defp do_wait_until_tx_sync_to_watcher(_session, _tx_id, 0), do: :wait_until_tx_sync_failed
+
+  defp do_wait_until_tx_sync_to_watcher(session, tx_id, retry) do
+    {:ok, response} =
+      Api.Transaction.transaction_get(
+        Connection.client(),
+        %Model.GetTransactionBodySchema{
+          id: tx_id
+        }
+      )
+
+    if Jason.decode!(response.body)["success"] do
```

Comment:
this is a somewhat unusual usage of if cases.
https://erlef.github.io/security-wg/secure_coding_and_deployment_hardening/elixir_truthy
in 99% of  cases, you should stick with `case` statements in elixir.

Reply:
```
Interactive Elixir (1.8.2) - press Ctrl+C to exit (type h() ENTER for help)
iex(1)> if true, do: 1 + 2
3
iex(2)> if "true", do: 1 + 2
3
iex(3)> if "false", do: 1 + 2
3
```
I think that `Jason.decode!(response.body)["success"]` returns either "true" or "false", correct? they're not atoms I think.